src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2011-2015, jcabi.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.http.wire;
+
+import com.jcabi.http.Request;
+import com.jcabi.http.Response;
+import com.jcabi.http.Wire;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Wire that caches requests based on Last-Modified
+ * and If-Modified-Since headers.
+ * @author Igor Piddubnyi (igor.piddubnyi@gmail.com)
+ * @version $Id$
+ * @since 1.15
+ */
+public final class LastModifiedCachingWire implements Wire {
+
+    /**
+     * Last-Modified header name.
+     */
+    public static final String LAST_MODIFIED = "Last-Modified";
+
+    /**
+     * If-Modified-Since header name.
+     */
+    private static final String IF_MODIFIED_SINCE = "If-Modified-Since";
+
+    /**
+     * Cache.
+     */
+    private final transient Map<Request, Response> cache;
+
+    /**
+     * Original wire.
+     */
+    private final transient Wire origin;
+
+    /**
+     * Public ctor.
+     * @param wire Original wire
+     */
+    public LastModifiedCachingWire(final Wire wire) {
+        this.origin = wire;
+        this.cache = new ConcurrentHashMap<>();
+    }
+
+    // @checkstyle ParameterNumber (5 lines)
+    @Override
+    public Response send(final Request req, final String home,
```

Comment:
What if the original incoming request already has an `If-Modified-Since` header?
Perhaps it should bypass the cache?